telegram/__init__.py

```diff
@@ -160,40 +172,109 @@ def update_item(self, item, caller=None, source=None, dest=None):
             # for cid in self._chat_ids:
             #    self._bot.sendMessage(cid, msg_txt)
 
-    def _msg_broadcast(self, msg):
-        for cid in self._chat_ids:
+    def _msg_broadcast(self, msg, chat_id=None):
+        chat_ids_to_send = []
+        if chat_id is None:
+            chat_ids_to_send = self._chat_ids
+        else:
+            if isinstance(chat_id, list):
+                chat_ids_to_send = chat_id
+            else:
+                chat_ids_to_send.append(chat_id)
+        for cid in chat_ids_to_send:
             try:
                 self._bot.sendMessage(cid, msg)
             except:
                 self.logger.error("could not broadcast to chat id [%d]" % cid)
+                
+    def _photo_broadcast(self, photofile, msg, chat_id=None):
+        chat_ids_to_send = []
+        if chat_id is None:
+            chat_ids_to_send = self._chat_ids
+        else:
+            if isinstance(chat_id, list):
+                chat_ids_to_send = chat_id
+            else:
+                chat_ids_to_send.append(chat_id)
```

Comment:
Would it be possible to extract this part of code (checking for chat_id and constructing a list, line 191-198) into an own method and use it also at line 177?